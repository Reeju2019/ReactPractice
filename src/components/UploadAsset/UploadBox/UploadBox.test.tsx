import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import { UploadBox } from "."

const localeConfig = {
  uploadingPercentage: 100,
  uploadMsg: "Hello world",
}

describe("UploadBox Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <UploadBox
        uploadingPercentage={localeConfig.uploadingPercentage}
        uploadMsg={localeConfig.uploadMsg}
        uploadCancelHandler={() => {}}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <UploadBox
        uploadingPercentage={localeConfig.uploadingPercentage}
        uploadMsg={localeConfig.uploadMsg}
        uploadCancelHandler={() => {}}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
