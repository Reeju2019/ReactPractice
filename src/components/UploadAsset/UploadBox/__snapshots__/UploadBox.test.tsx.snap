// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadBox Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Row
      className="show-progressbar-box align-items-center justify-content-center py-3"
    >
      <Col
        md={4}
        xs={2}
      >
        Hello world
      </Col>
      <Col
        md={2}
        xs={2}
      >
        100%
      </Col>
      <Col
        md={4}
        xs={2}
      >
        <ProgressBar
          animated={false}
          isChild={false}
          max={100}
          min={0}
          now={100}
          striped={false}
          visuallyHidden={false}
        />
      </Col>
      <Col
        md={2}
        xs={2}
      />
    </Row>
  </React.Fragment>,
]
`;
